cmd/version/version.go

```diff
@@ -60,23 +66,62 @@ func getVersion(ctx context.Context, args []string, cmdCtx cmdCore.CommandContex
 		return err
 	}
 
+	getControlePlaneVersion(ctx, cmdCtx)
+	return nil
+}
+
+func printVersion(response versionOutput) error {
+	b, err := json.MarshalIndent(response, "", "  ")
+	if err != nil {
+		return fmt.Errorf("err %v: ", err)
+	}
+	fmt.Print(string(b))
+	return nil
+}
+
+func compareVersion(latest, current string) string {
+	latestVersion, _ := hversion.NewVersion(latest)
+	fversion, _ := hversion.NewVersion(current)
+
+	if fversion.LessThan(latestVersion) {
+		return fmt.Sprintf("A newer version of flytectl is available [%v] Please upgrade using - https://docs.flyte.org/projects/flytectl/en/latest/index.html", latest)
+	}
+
+	return "Installed flytectl version is the latest"
+}
+
+func getControlePlaneVersion(ctx context.Context, cmdCtx cmdCore.CommandContext) {
+	v, err := cmdCtx.AdminClient().GetVersion(ctx, &admin.GetVersionRequest{})
+	if err != nil || v == nil {
+		logger.Debugf(ctx, "Failed to get version of control plane %v: \n", err)
+		return
+	}
 	// Print Flyteadmin
 	if err := printVersion(versionOutput{
 		Build:     v.ControlPlaneVersion.Build,
 		BuildTime: v.ControlPlaneVersion.BuildTime,
 		Version:   v.ControlPlaneVersion.Version,
 		App:       "controlPlane",
 	}); err != nil {
-		return err
+		logger.Debugf(ctx, "Not able to get control plane version..Please try again : %v \n", err)
 	}
-	return nil
 }
 
-func printVersion(response versionOutput) error {
-	b, err := json.MarshalIndent(response, "", "  ")
+func getLatestVersion(baseURL string) (string, error) {
+	response, err := http.Get(fmt.Sprintf("%v/repos/flyteorg/flytectl/releases/latest", baseURL))
 	if err != nil {
-		return fmt.Errorf("err %v: ", err)
+		return "", err
 	}
-	fmt.Print(string(b))
-	return nil
+	defer response.Body.Close()
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", err
+	}
+	var result = make(map[string]interface{})
```

Comment:
Can this not be map[string]string. Do we expect any other type for any of the Version responses?

Reply:
ok i will create proper struct